public/src/components/promoTool/createPromoCampaignDialog.tsx

```diff
@@ -53,9 +76,22 @@ const CreatePromoCampaignDialog: React.FC<CreatePromoCampaignDialogProps> = ({
         </IconButton>
       </div>
       <DialogContent dividers>
-        <ProductSelector />
+        <ProductSelector
+          selectedValue={selectedProduct.toString()}
+          handleSelectedValue={setSelectedProduct}
+        />
         <TextField
           className={classes.input}
+          error={errors.name !== undefined}
+          helperText={errors.name ? errors.name.message : NAME_DEFAULT_HELPER_TEXT}
+          {...register('name', {
+            required: EMPTY_ERROR_HELPER_TEXT,
+            pattern: {
+              value: VALID_CHARACTERS_REGEX,
+              message: INVALID_CHARACTERS_ERROR_HELPER_TEXT,
+            },
+            // validate: createDuplicateValidator(existingNames), TODO: would this be useful?
```

Comment:
I know the PromoCampaign name does not *need* to be unique, however, would it be useful if it were not duplicated? I can see the potential for some confusion if there are two different campaigns with the same name.